addons/payment_paypal/models/payment_transaction.py

```diff
@@ -78,14 +78,20 @@ def _get_specific_processing_values(self, processing_values):
             "Sending '/checkout/orders' request for transaction with reference %s:\n%s",
             self.reference, pprint.pformat(payload)
         )
+        idempotency_key = payment_utils.generate_idempotency_key(
+            self, scope='payment_request_order'
+        )
         order_data = self.provider_id._paypal_make_request(
-            '/v2/checkout/orders', json_payload=payload
+            '/v2/checkout/orders', json_payload=payload, idempotency_key=idempotency_key
         )
         _logger.info(
             "Response of '/checkout/orders' request for transaction with reference %s:\n%s",
             self.reference, pprint.pformat(order_data)
         )
-        return {'order_id': order_data['id']}
+        return {
+            'order_id': order_data['id'],
+            'tx_id': self.id,
```

Comment:
I think it should have been part of the generic processing values already, but since it's not, let's keep things simple and rely on the reference instead, which is already in the processing values and can be sent by the client during the RPC.